### ISSUE TITLE
App crashes

### ISSUE THREAD

**Issue by wolkenschieber** [blokadaorg/blokada#52]
Blokada 2.2.2 crashes, if I
- open about dialog
- tab on *Karsen*

**Comment by kar** [comment]
Which device?

**Comment by wolkenschieber** [comment]
The device is a Moto G4. The app was installed from fdroid.

**Comment by wolkenschieber** [comment]
Logcat shows following exception when the crash happens:
```
09-23 13:57:25.684 18397 18397 E AndroidRuntime: FATAL EXCEPTION: main
09-23 13:57:25.684 18397 18397 E AndroidRuntime: Process: org.blokada.alarm, PID: 18397
09-23 13:57:25.684 18397 18397 E AndroidRuntime: android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.app.ContextImpl.startActivity(ContextImpl.java:672)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.app.ContextImpl.startActivity(ContextImpl.java:659)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.content.ContextWrapper.startActivity(ContextWrapper.java:331)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.text.style.URLSpan.onClick(URLSpan.java:75)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.text.method.LinkMovementMethod.onTouchEvent(LinkMovementMethod.java:217)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.widget.TextView.onTouchEvent(TextView.java:8337)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.View.dispatchTouchEvent(View.java:9303)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2555)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at com.android.internal.policy.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2403)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1737)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.app.Activity.dispatchTouchEvent(Activity.java:2775)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.support.v7.view.i.dispatchTouchEvent(Unknown Source)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2364)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.View.dispatchPointerEvent(View.java:9523)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4242)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4108)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3654)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3707)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3673)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3799)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3681)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3856)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3654)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3707)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3673)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3681)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3654)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:5969)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:5908)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:5869)
09-23 13:57:25.684 18397 18397 E AndroidRuntime:        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(Vie
09-23 13:57:25.688  1995  2066 W ActivityManager:   Force finishing activity org.blokada.alarm/org.blokada.ui.app.android.MainActivity
```

**Comment by wolkenschieber** [comment]
I really wanted to fix this issue and submit a pull request, but I failed miserably. 

It looked promissing as the issue is tagged *simple*, which matches my Kotlin knowledge. Furthermore the error message indicated, that just a flag in an intent needed to be set. Turns out it, it is not.

The exception occurs when opening an HTML link from text in `AUpdateView.creditsView`. There's no intent involved directly. Research, Google search and stackoverflow posts, indicate that an unsuitable `Context` might be used here, but I doubt that.

Furthermore the issue occurs on an AVD using API 23, but not on one using API 27. So there's also API level bugfixing involved. And again my Google-fu digs up nothing here.

I'm out of options, but I'll keep following this issue as I'm curious about how this issue will be fixed.
